client/implementation/src/main/java/io/smallrye/graphql/client/typesafe/impl/reflection/ParameterInfo.java

```diff
@@ -45,6 +47,10 @@ private String graphQlInputTypeName(TypeInfo type) {
     }
 
     private String simpleInputTypeName(TypeInfo type) {
+        if (type.isAnnotated(Input.class))
+            return type.getAnnotation(Input.class).value();
+        if (type.isAnnotated(Type.class))
```

Comment:
Input should not use `@Type`. First look `@Input`, then `@Name` then `className`. On the server we also have a NamingStrategy when using ClassName. Default is just the ClassName, then you also MergeInnerClass (so Inner class will do parentClassNameInnerClassName, Or you can do Full then it use the full packageName. Naming Strategy can be set via config.

Reply:
You're absolutely right. I just fixed that.